gr-digital/lib/additive_scrambler_bb_impl.cc

```diff
@@ -56,25 +56,26 @@ additive_scrambler_bb_impl::additive_scrambler_bb_impl(int mask,
 
 additive_scrambler_bb_impl::~additive_scrambler_bb_impl() {}
 
-int additive_scrambler_bb_impl::mask() const { return d_lfsr.mask(); }
+uint64_t additive_scrambler_bb_impl::mask() const { return d_lfsr.mask(); }
 
-int additive_scrambler_bb_impl::seed() const { return d_seed; }
+uint64_t additive_scrambler_bb_impl::seed() const { return d_seed; }
 
-int additive_scrambler_bb_impl::len() const { return d_len; }
+uint8_t additive_scrambler_bb_impl::len() const { return d_len; }
 
-int additive_scrambler_bb_impl::count() const { return d_count; }
+int64_t additive_scrambler_bb_impl::count() const { return d_count; }
 
-int additive_scrambler_bb_impl::_get_next_reset_index(int noutput_items,
-                                                      int last_reset_index /* = -1 */)
+int64_t
+additive_scrambler_bb_impl::_get_next_reset_index(int64_t noutput_items,
+                                                  int64_t last_reset_index /* = -1 */)
 {
-    int reset_index =
+    int64_t reset_index =
         noutput_items; // This is a value that gets never reached in the for loop
     if (d_count == -1) {
```

Comment:
Can’t we use `d_count = 0` as a special value and consequently `uint64_t` instead of `int64_t` for `d_count`? Resetting the scrambler after 0 Bytes doesn't make sense.

Reply:
You are possibly correct, I haven't looked in to this. This block had a bug that was fix due to this weird usage of the d_count, so I'm keen to correct this aswell, there is no point in waiting, I guess.

Reply:
The only thing I can think of is that d_count=0 will disable the tags-section. If that is desired or planned, I don't know. But I don't see any issue with using int64_t either though.

Reply:
Will resolve this conversation, since there is no problem here. I do see a potential fix for this in the future, so it's not that it is a bad idea, but not just part of this commit.